azure/Guidelines.md

```diff
@@ -1292,6 +1292,52 @@ In addition to distributed tracing, Azure also uses a set of common correlation
 |x-ms-client-request-id       |Both      |Optional. Caller-specified value identifying the request, in the form of a GUID with no decoration such as curly braces (e.g. `x-ms-client-request-id: 9C4D50EE-2D56-4CD3-8152-34347DC9F2B0`). If the caller provides this header the service **must** include this in their log entries to facilitate correlation of log entries for a single request. Because this header can be client-generated, it should not be assumed to be unique by the service implementation.
 |x-ms-request-id              |Response  |Required. Service generated correlation id identifying the request, in the form of a GUID with no decoration such as curly braces. In contrast to the the `x-ms-client-request-id`, the service **must** ensure that this value is globally unique. Services should log this value with their traces to facilitate correlation of log entries for a single request.
 
+<a href="#RP-API-MSD" name="RP-API-MSD"></a>
+### RP API More Secure Defaults Guidance
+
+The RP API MSD Guidelines are designed to ensure that any new RP API being developed is deployed from the outset with secure parameter default values. This proactive approach helps prevent non-secure defaults and maintains a high security standard from the very beginning of the API development process.
```

Comment:
...either more secure default parameter values or making the previously optional parameter as required"

Reply:
Defaulting vs. explicitness is a difficult trade-off. 

Default values allow the service to improve its security posture over time with new capabilities. Having clients explicitly "pin" values can slow down adoption of new capabilities. Adding new required parameters means breaking changes. Which slows down adoption of new api versions. Which in turn slows down the ability to use the new features. 

In practical terms, we need to have a default value for new security related properties when we migrate previously existing resources. That doesn't have to mean that we use the same default values in the public api definitions, but it doesn't seem like an unreasonable thing to do either.

Reply:
I have had numerous conversations about services exposing default values for properties and have been meaning to write a guideline about it. The guideline would go something like this: when a structure/model is first introduced by a service (V1 of structure), very few (if any) of the properties should have default values. Customer must explicitly select values for the properties. This prevents breaking changes should the service want to change a default since there are no defaults. Any new properties that are added to the structure/model in the future (V2+) must have default values in order to avoid breaking changes.